src/m365/pp/commands/solution/solution-get.ts

```diff
@@ -0,0 +1,125 @@
+    return commands.SOLUTION_GET;
+  }
+
+  public get description(): string {
+    return 'Lists a specific solution in a given environment.';
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['uniquename', 'version', 'publisher'];
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initOptionSets();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        asAdmin: !!args.options.asAdmin
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-e, --environment <environment>'
+      },
+      {
+        option: '-i, --id'
+      },
+      {
+        option: '-n, --name'
+      },
+      {
+        option: '-a, --asAdmin'
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push(
+      ['id', 'name']
+    );
+  }
+
+  public async commandAction(logger: Logger, args: any): Promise<void> {
+    if (this.verbose) {
+      logger.logToStderr(`Retrieving a specific solutions for which the user is an admin...`);
+    }
+
+    try {
+      const res: Solution = await this.getSolution(args.options);
+      if (!args.options.output || args.options.output === 'json') {
+        logger.log(res);
+      }
+      else {
+        //converted to text friendly output
+        logger.log({
+          uniquename: res.uniquename,
+          version: res.version,
+          publisher: (res.publisherid as Publisher).friendlyname
+        });
+      }
+    }
+    catch (err: any) {
+      this.handleRejectedODataJsonPromise(err);
+    }
+  }
+
+  private async getSolution(options: Options): Promise<Solution> {
+    const dynamicsApiUrl = await powerPlatform.getDynamicsInstanceApiUrl(options.environment, options.asAdmin);
+    if (options.id) {
+      const requestOptions: AxiosRequestConfig = {
+        url: `${dynamicsApiUrl}/api/data/v9.0/solutions(${options.id})?$expand=publisherid($select=friendlyname)&$select=solutionid,uniquename,version,publisherid,installedon,solutionpackageversion,friendlyname,versionnumber&api-version=9.1`,
+        headers: {
+          accept: 'application/json;odata.metadata=none'
+        },
+        responseType: 'json'
+      };
+
+      const r: Solution = await request.get<Solution>(requestOptions);
+      return r;
+    }
+
+    const requestOptions: AxiosRequestConfig = {
+      url: `${dynamicsApiUrl}/api/data/v9.0/solutions?$filter=isvisible eq true and uniquename eq \'${options.name}\'&$expand=publisherid($select=friendlyname)&$select=solutionid,uniquename,version,publisherid,installedon,solutionpackageversion,friendlyname,versionnumber&api-version=9.1`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    const r = await request.get<{ value: Solution[] }>(requestOptions);
+    return r.value[0];
```

Comment:
If no solution was returned, we should throw an error, something like
`The specified solution '${name}' does not exist.`